### PR TITLE
Generate code coverage reports

### PR DESCRIPTION
**Pull request by mrfelton**
We have a test suite, so we should have code coverage reports to go with it.

Coverage reports make it easy to see what is and isn't covered by tests and encourages developers to write tests for new and existing code.

This PR:

 - generates test coverage reports whenever tests are run (output to `.coverage` directory)
 - provides the command `yarn coverage` which will open up an html version of the test coverage report (see screenshot below)
 - posts test coverage data to coveralls (eg https://coveralls.io/github/mrfelton/zap-desktop) at the end of every successful build on travis
 - adds a status badge to the project homepage to provide visibility into he current test coverage of the project (I also added a couple of other useful status badges).

image
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
-<h1 align='center'>
-  <img src='https://imgur.com/svn8Jrw.jpg' alt='screenshot' />
-  <br />
-  <center>
-    <a href='https://zap.jackmallers.com'>Zap</a>
-  </center>
-</h1>
+<p>
+  <h1 align='center'>
```
[the remark this answers is not in the document]
Comment (mrfelton) [reply README.md]: Cheers. Fixed.

**Comment by Empact** [comment]
Quite a quick response! Checked your branch and saw the badges aren't rendering properly:
https://github.com/mrfelton/zap-desktop/tree/feature/test-coverage

**Comment by mrfelton** [comment]
They don’t render properly because the links point to the LN-Zap github user account - which needs to be connected to coveralls. (I tested with my own github repo and it works fine, as per the screenshot I attached)

--
Tom Kirkpatrick
Founder & CTO, KirkDesigns
+34 (0) 681 395 967 | Skype: tkirkpatrick99

**Comment by mrfelton** [comment]
@jackmallers would you be able to connect this repo to [Coveralls](https://coveralls.io/) so that we can make some test coverage reports available?

**Comment by mrfelton** [comment]
rebased

**Comment by Empact** [comment]
utACK 991e1a0

**Pull request merged**